patches/pcl_ros.patch

```diff
@@ -1,35 +1,33 @@
+-target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
++#add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
++#target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
  
 -add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
--target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-+# add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
-+# target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+-target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
++#add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
++#target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
  
 -add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
--target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-+# add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
-+# target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+-target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
++#add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
++#target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARY_DIRS} ${PCL_LIBRARIES})
  
 -add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
--target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-+# add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
-+# target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+-target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
++#add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
++#target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
  
 -add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
--target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-+# add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
-+# target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+-target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
++#add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
++#target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${EIGEN3_LIBRARIES} ${PCL_LIBRARIES})
```

Comment:
> Why would you require the ${EIGEN3_LIBRARY_DIRS} and ${EIGEN3_LIBRARIES}?

That's a good question indeed. 
In any case I had a different issue with the executables: a `make` error when linking against `rosbag_storage` and an undefined symbol from `bzip2`. In any case it seemed easier to just skip these executables.

Reply:
Yes, I have been looking into this last week. I am in the process of preparing and submitting individual PRs.

Many additional patches are required because library dependencies are not transitive for static libraries. So all packages need to properly export their internal library dependencies.

Reply:
See #27.